### PR TITLE
Added 3 new tests for the maintenance window

### PR DESCRIPTION
**Pull request by dgarc330**
Added def test_011_delete_mw_on_switch(self) function

### CODE REVIEW

File: tests/test_e2e_15_maintenance.py [hunk 1]
```diff
@@ -120,3 +120,38 @@ def test_010_create_mw_on_switch_should_move_evc(self):
...
+            "start": start.strftime(TIME_FMT),
+            "end": end.strftime(TIME_FMT),
+            "items": [
+                "00:00:00:00:00:00:02"
+            ]
+        }
+
+        api_url = KYTOS_API + '/maintenance/'
+        # 1 Send get request to API to get maintenance schema
+        response = requests.get(api_url, data=json.dumps(payload), headers={'Content-type': 'application/json'})["id"]
```
Comment (ajoaoff) [inline tests/test_e2e_15_maintenance.py:137]: Here you should make a POST request, not a GET, to create the maintenance window.
```suggestion
        response = requests.post(api_url, data=json.dumps(payload), headers={'Content-type': 'application/json'})["id"]
```

File: tests/test_e2e_15_maintenance.py [hunk 2]
```diff
@@ -120,3 +120,38 @@ def test_010_create_mw_on_switch_should_move_evc(self):
...
+        mw_api_url = KYTOS_API + '/maintenance/' + mw_id
+        delete_response = requests.delete(mw_api_url, data=json.dumps(payload), headers={'Content-type': 'application/json'})
+        # 4. Verify that code 204(success) is given back
+        assert delete_response.status_code == 204
+        # 5. Request list of maintenance windows again, and verify that the deleted mw_id is not on maintenance list
+        req = requests.get(api_url, data=json.dumps(payload), headers={'Content-type': 'application/json'})
+        json_data = req.json()
+        for attribute in json_data:
+            if attribute['id'] != mw_id:
+                print("Maintenance window deleted correctly")
```
Comment (ajoaoff) [inline tests/test_e2e_15_maintenance.py:151]: In a GET request you should not provide a payload.
You could also request for the specific maintenance window you are going to check, and see if the status is 404.
```suggestion
        req = requests.get(my_api_url)
        assert req.status_code == 404
```

**Review by ajoaoff — CHANGES_REQUESTED** [review]

**Review by ajoaoff — APPROVED** [review]

File: tests/test_e2e_15_maintenance.py [hunk 3]
```diff
@@ -120,3 +120,33 @@ def test_010_create_mw_on_switch_should_move_evc(self):
...
+            "start": start.strftime(TIME_FMT),
+            "end": end.strftime(TIME_FMT),
+            "items": [
+                "00:00:00:00:00:00:02"
+            ]
+        }
+
+        api_url = KYTOS_API + '/maintenance/'
+        # 1 Send get request to API to get maintenance schema
+        response = requests.get(api_url, data=json.dumps(payload), headers={'Content-type': 'application/json'})["id"]
```
[the remark this answers is not in the document]
Comment (dgarc330) [reply tests/test_e2e_15_maintenance.py]: You are right Italo, that is leftover code. I will update the commit to remove that, thank you

File: tests/test_e2e_15_maintenance.py [hunk 4]
```diff
@@ -120,3 +120,35 @@ def test_010_create_mw_on_switch_should_move_evc(self):
...
+            "start": start.strftime(TIME_FMT),
+            "end": end.strftime(TIME_FMT),
+            "items": [
+                "00:00:00:00:00:00:02"
+            ]
+        }
+
+        api_url = KYTOS_API + '/maintenance/'
+        # 1 Send get request to API to get maintenance schema
+        response = requests.post(api_url, data=json.dumps(payload), headers={'Content-type': 'application/json'})["id"]
```
Comment (italovalcy) [inline tests/test_e2e_15_maintenance.py:137]: The id should be removed from the end of the request.post call:

```suggestion
        response = requests.post(api_url, data=json.dumps(payload), headers={'Content-type': 'application/json'})
```

**Review by italovalcy — CHANGES_REQUESTED** [review]
Daniela, here is my review requesting the changes

File: tests/test_e2e_15_maintenance.py [hunk 5]
```diff
@@ -120,3 +120,35 @@ def test_010_create_mw_on_switch_should_move_evc(self):
         h11.cmd('ip link del vlan100')
         h3.cmd('ip link del vlan100')
 
+    def test_011_delete_mw_on_switch(self):
+        # 0. Start maintenance window
+        start = datetime.now() + timedelta(seconds=60)
+        end = start + timedelta(seconds=60)
+        payload = {
```
Comment (italovalcy) [inline tests/test_e2e_15_maintenance.py:127]: Even though a description is not mandatory for a MW, due to an issue (https://github.com/kytos/maintenance/issues/28) the test will not work if you dont provide the description

```suggestion
        payload = {
            "description":  "mw for test 1",
```

File: tests/test_e2e_15_maintenance.py [hunk 6]
```diff
@@ -120,3 +120,35 @@ def test_010_create_mw_on_switch_should_move_evc(self):
...
+                "00:00:00:00:00:00:02"
+            ]
+        }
+
+        api_url = KYTOS_API + '/maintenance/'
+        # 1 Send get request to API to get maintenance schema
+        response = requests.post(api_url, data=json.dumps(payload), headers={'Content-type': 'application/json'})["id"]
+        # 2. from the output of the GET request, extract the mw_id
+        json_data = response.json()
+        mw_id = json_data["id"]
```
Comment (italovalcy) [inline tests/test_e2e_15_maintenance.py:140]: the correct returned attribute is `mw_id` not `id`
```suggestion
        mw_id = json_data["mw_id"]
```

File: tests/test_e2e_15_maintenance.py [hunk 7]
```diff
@@ -120,3 +120,35 @@ def test_010_create_mw_on_switch_should_move_evc(self):
...
+        # 3. Provide mw_id to API call to delete said mw_id
+        mw_api_url = KYTOS_API + '/maintenance/' + mw_id
+        delete_response = requests.delete(mw_api_url, data=json.dumps(payload), headers={'Content-type': 'application/json'})
+        # 4. Verify that code 204(success) is given back
+        assert delete_response.status_code == 204
```
Comment (italovalcy) [inline tests/test_e2e_15_maintenance.py:145]: The return code is actually 200 not 204
```suggestion
        assert delete_response.status_code == 200
```

File: tests/test_e2e_15_maintenance.py [hunk 8]
```diff
@@ -120,3 +120,35 @@ def test_010_create_mw_on_switch_should_move_evc(self):
...
+        # 5. Request list of maintenance windows again, and verify that the deleted mw_id is not on maintenance list
+        req = requests.get(my_api_url)
```
Comment (italovalcy) [inline tests/test_e2e_15_maintenance.py:147]: typo:
```suggestion
        req = requests.get(mw_api_url)
```

**Review by italovalcy — CHANGES_REQUESTED** [review]

**Pull request closed without merging**